lib/kadai-core/src/main/java/io/kadai/task/internal/TaskCommentServiceImpl.java

```diff
@@ -140,6 +146,51 @@ TaskComment createTaskComment(TaskComment taskCommentToCreate)
     return taskCommentImplToCreate;
   }
 
+  public BulkOperationResults<String, KadaiException> createTaskCommentsBulk(
+          Collection<String> taskIds, String text) throws InvalidArgumentException {
+
+    if (taskIds == null) {
+      throw new InvalidArgumentException("taskIds must not be null");
+    }
+    if (text == null || text.isEmpty()) {
+      throw new InvalidArgumentException("text must not be null/empty");
+    }
+
+    kadaiEngine.openConnection();
+    try {
+      BulkOperationResults<String, KadaiException> errors = new BulkOperationResults<>();
+      Instant now = Instant.now();
+
+      Set<String> existingTaskIds = taskMapper.findExistingTasks(taskIds, null).stream()
+              .map(MinimalTaskSummary::getTaskId)
+              .collect(Collectors.toSet());
+
+      taskIds.stream()
+              .filter(id -> !existingTaskIds.contains(id))
+              .forEach(id -> errors.addError(
+                      id, new TaskNotFoundException("Task not found: " + id)));
```

Comment:
Careful, the constructor takes the task-id, not a human-readable message.
```suggestion
                      id, new TaskNotFoundException(id)));
```